modules/test/playwright/tests/portal-web/html/taglib/ui/pageIterator.spec.ts

```diff
@@ -114,3 +126,136 @@ test(
+		await pageEditorPage.goto(layout, site.friendlyUrlPath);
+
+		await pageEditorPage.goToWidgetConfiguration(widgetId);
+
+		const assetSelectionTab = page
+			.frameLocator('iframe[title="Configuration"]')
+			.getByRole('tab', {name: 'Asset Selection'});
+		await assetSelectionTab.waitFor({state: 'visible'});
+		await assetSelectionTab.click();
+
+		await clickAndExpectToBeVisible({
+			autoClick: true,
+
+			target: page
+				.frameLocator('iframe[title="Configuration"]')
+				.frameLocator('iframe[title="Select Collection"]')
+				.getByRole('link', {name: 'Collection Providers'}),
+			timeout: 2000,
+			trigger: page
+				.frameLocator('iframe[title="Configuration"]')
+				.getByRole('button', {exact: true, name: 'Select Collection'}),
+		});
+
+		await clickAndExpectToBeHidden({
+			target: page
+				.frameLocator('iframe[title="Configuration"]')
+				.locator('.modal-dialog'),
+			timeout: 2000,
+			trigger: page
+				.frameLocator('iframe[title="Configuration"]')
+				.frameLocator('iframe[title="Select Collection"]')
+				.getByRole('button', {name: 'Select Recent Content'}),
+		});
+
+		await page
+			.frameLocator('iframe[title="Configuration"]')
+			.getByRole('tab', {name: 'Display Settings'})
+			.click();
+
+		const itemDisplayInput = page
+			.frameLocator('iframe[title="Configuration"]')
+			.getByLabel('Number of Items to Display');
+
+		await itemDisplayInput.waitFor({state: 'visible'});
+
+		await itemDisplayInput.click();
+
+		await itemDisplayInput.fill('1');
+
+		await page
+			.frameLocator('iframe[title="Configuration"]')
+			.getByLabel('Pagination Type')
+			.selectOption('Regular');
+
+		await page
+			.frameLocator('iframe[title="Configuration"]')
+			.getByRole('button', {name: 'Save'})
+			.click();
+
+		await page
+			.frameLocator('iframe[title="Configuration"]')
+			.getByRole('button', {name: 'Cancel'})
+			.click();
+
+		await page.getByLabel('Publish', {exact: true}).click();
+
+		await page.goto(
+			`/web${site.friendlyUrlPath}/${layout.friendlyUrlPath}`
+		);
+
+		for (let i = 1; i <= 10; i++) {
+			const contentStructureId =
+				await getBasicWebContentStructureId(apiHelpers);
+			const randomTitle = getRandomString();
+
+			const webContent =
+				await apiHelpers.jsonWebServicesJournal.addWebContent({
+					ddmStructureId: contentStructureId,
+					groupId: site.id,
+					titleMap: {en_US: randomTitle},
+				});
+
+			apiHelpers.data.push({
+				id: `${site.id}_${webContent.articleId}`,
+				type: 'webContent',
+			});
+		}
+
+		await page.reload();
+
+		await page.evaluate(() => {
+			const intermediatePagesButton = document.querySelector(
+				'button[title="Show Intermediate Pages"]'
+			);
+
+			if (intermediatePagesButton) {
+				(intermediatePagesButton as HTMLElement).click();
+			}
+		});
+
+		const elementHandle = await page.evaluateHandle(() => {
+			return document.querySelector(
+				'.dropdown-menu-top.dropdown-menu-width-shrink'
+			);
+		});
+
+		await expect(elementHandle).toBeTruthy();
```

Comment:
@fortunatomaldonado This is technically fine I guess, but it's not really a test from feature perspective. Accurate test would be
1. Set resolution to small
2. Open paginatior dropdown
3. Asset an option in dropdown is visible

Reply:
Let me know if the new changes are okay.